spec/lib/msf/core/exploit/remote/remote/http/http_cookie_spec.rb

```diff
@@ -0,0 +1,815 @@
+
+      it 'false passed to secure is set as false' do
+        f = false
+
+        cookie.secure = f
+
+        expect(cookie.secure).to eql(false)
+      end
+    end
+  end
+
+  describe 'created_at' do
+    describe 'Time' do
+      it 'is returned by created_at after the cookie is initialized' do
+        # cookie created in before
+
+        c = cookie.created_at
+
+        expect(c.class).to eql(Time)
+      end
+
+      it 'instance is assigned to created_at successfully' do
+        t = Time.now
+
+        cookie.created_at = t
+
+        expect(cookie.created_at).to eql(t)
+        expect(cookie.created_at.class).to eql(Time)
+      end
+    end
+
+    describe 'Non-Time' do
+      it 'which can be parsed to a Time object, will be assigned to created_at successfully' do
+        obj = Time.now.to_s
+
+        cookie.created_at = obj
+
+        expect(cookie.created_at).to eql(Time.parse(obj))
+      end
+
+      it 'which cannot be parsed to a Time object, when assigned to created_at, will throw a TypeError' do
+        obj = [rand(9), rand(9), rand(9)]
+
+        expect do
+          cookie.created_at = obj
+        end.to raise_error(TypeError)
+      end
+    end
+
+    describe 'nil' do
+      it 'assigned to created_at successfully' do
+        cookie.created_at = nil
+        expect(cookie.created_at).to eql(nil)
+      end
+    end
+  end
+
+  describe 'accessed_at' do
+    describe 'Time' do
+      it 'is returned by accessed_at after the cookie is initialized' do
+        # cookie created in before
+
+        c = cookie.accessed_at
+
+        expect(c.class).to eql(Time)
+      end
+
+      it 'instance is assigned to accessed_at successfully' do
+        t = Time.now
+
+        cookie.accessed_at = t
+
+        expect(cookie.accessed_at).to eql(t)
+        expect(cookie.accessed_at.class).to eql(Time)
+      end
+    end
+
+    describe 'Non-Time' do
+      it 'which can be parsed to a Time object, will be assigned to accessed_at successfully' do
+        obj = Time.now.to_s
+
+        cookie.accessed_at = obj
+
+        expect(cookie.accessed_at).to eql(Time.parse(obj))
+      end
+
+      it 'which cannot be parsed to a Time object, when assigned to accessed_at, will throw a TypeError' do
+        obj = [rand(9), rand(9), rand(9)]
+
+        expect do
+          cookie.accessed_at = obj
+        end.to raise_error(TypeError)
+      end
+    end
+
+    describe 'nil' do
+      it 'assigned to accessed_at successfully' do
+        cookie.accessed_at = nil
+        expect(cookie.accessed_at).to eql(nil)
+      end
+    end
+  end
+
+  describe 'session' do
+    describe 'Session' do
+      it 'is set as True when initialized with no max_age or expires args and a nil value' do
+        # cookie created in before
+
+        s = cookie.session?
+
+        expect(s).to eql(true)
+      end
+
+      it 'is set as the nil? value of max_age when a valid value is assigned to max_age' do
+        max_age = rand(0..1) == 1 ? 1 : nil
```

Comment:
I don't get why you're doing this, can't you just have two tests? one where max is is nil and one where it isn't?